### ISSUE TITLE
Is it possible to vectorize the input of the odeint function?

### ISSUE THREAD

**Issue by Antonio-Nappi** [scipy/scipy#13348]
Hello everybody, I want to know if it's possible to vectorize the odeint function because I execute it very often with different parameters. If could it be possible I will save a lot of time.

**Comment by TejasAvinashShetty** [comment]
Could you give an example as to how you intend to use the vectorization if available?

**Comment by Antonio-Nappi** [comment]
Since I have to solve the same differential equation with different sets of parameters, I'd like know if it's possibile something like:
y = odeint(function,x0,args=(p,)) where x0 is a vector (N,M) with each row a different initial condition, p is a vector (N,K) with each row a different set of parameters, and y is a vector(N,M) where each row is the output of the corresponding odeint function

**Comment by ilayn** [comment]
Not out-of-the-box I think unfortunately. Since ode solutions are recursive, I don't think you would save a massive amount of time if compared to a for loop. Only the function call overheads would be avoided even though sometimes that is already enough of a benefit. 

In any case, please consider `scipy.integrate.solve_ivp` instead if you can.

**Comment by Antonio-Nappi** [comment]
which are the main differences between `odeint` as `solve_ivp`?

**Comment by ilayn** [comment]
More variety of solvers and better control over the error. It can speed up the solutions greatly if your problem is well-conditioned.

**Comment by Antonio-Nappi** [comment]
ok, thank you, Is the function signature the same?

**Comment by ilayn** [comment]
Here is the documentation for all the options https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.solve_ivp.html

**Comment by laurent90git** [comment]
If each subproblem you want to to solve uses quite different time steps when solved on its own, you will likely not have much savings, if any. Indeed, the time step will be limited by the solution that has the most constraining error estimate at each step.
If that is the case, you may be better off solving each problem individually with a for loop. The for loop can be done in parallel, for example using the package "joblib".

**Comment by mdhaber** [comment]
Looks like the query has been answered. Also IIUC, I think you can do what you're looking for with two `ravel`s and two `reshape`s. The following solves for the trajectory (vertical movement only) of a particle under the influence of three different gravitational constants.  I've included the comments to indicate the lines of the `ravel`s and `reshape`s I'm referring to.

```python3
import matplotlib.pyplot as plt
import numpy as np
from scipy import integrate

x0 = np.zeros(3)
v0 = np.full(3, 30)
z0 = np.array([x0, v0])
preferred_shape = z0.shape

g = np.array([10, 20, 30])

def dzdt(t, z):
    z = np.reshape(z, preferred_shape)  # reshape
    dz = np.zeros_like(z)
    
    dz[0] = z[1]
    dz[1] = -g
    return dz.ravel()   # ravel

t_int = (0, 3)
res = integrate.solve_ivp(dzdt, t_int, z0.ravel(), dense_output=True)   # ravel
t = np.linspace(*t_int, 100)
z = res.sol(t).reshape(preferred_shape + (-1,))   # reshape
```
The point is that you can use whatever shapes you want inside and outside of your integrand; you just need to pass it through `solve_ivp` in 1D form.

Since this was phrased as a query, and since there are a few reasons provided here not to implement the corresponding feature, I'll go ahead and close this. But I wouldn't be opposed to seeing this feature, since similar arguments could be made against vectorizing `root_scalar` yet that tends to work quite well (see gh-7242). Feel free to open a new issue if you'd like to push for the enhancement.
